rest/admin_api_test.go

```diff
@@ -33,6 +33,121 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+// Test warnings being issued when a new channel is created with over 250 characters - CBG-1475
+func TestChannelNameSizeWarning(t *testing.T) {
+	syncFn := "function sync(doc, oldDoc) { channel(doc.chan); }"
+	rt := NewRestTester(t, &RestTesterConfig{
+		SyncFn: syncFn,
+	})
+	defer rt.Close()
+	getChannelNameWarnCount := func() int64 {
+		return rt.ServerContext().Database("db").DbStats.Database().WarnChannelNameSizeCount.Val
+	}
+
+	boundaryTestCases := []struct {
+		name               string
+		channelLengthToAdd int
+		expectWarn         bool
+	}{
+		{
+			name:               "Over max channel length",
+			channelLengthToAdd: 1,
+			expectWarn:         true,
+		},
+		{
+			name:               "Equal to max channel length",
+			channelLengthToAdd: 0,
+			expectWarn:         false,
+		},
+		{
+			name:               "Under max channel length",
+			channelLengthToAdd: -1,
+			expectWarn:         false,
+		},
+	}
+
+	boundaryTest := func(warnThresholdLength int) {
```

Comment:
Generally we try to avoid inline functions unless there is a really good reason for them as they make things harder to read. Especially as we're running the test cases  inside of it.